### ISSUE TITLE
master branch not compiling under SBT

### ISSUE THREAD

**Issue by sirthias** [propensive/magnolia#160]
While trying to put a PR together (with a simple performance fix) I stumbled across this compiler error with the current master branch:

```
[error] .../magnolia/examples/shared/src/main/scala/default.scala:31:28: no type parameters for method constructMonadic: (makeParam: magnolia.Param[magnolia.examples.HasDefault,T] => Monad[PType])(implicit monadic: mercator.Monadic[Monad])Monad[T] exist so that it can be applied to arguments (magnolia.Param[magnolia.examples.HasDefault,T] => scala.util.Either[String,magnolia.Param[magnolia.examples.HasDefault,T]#PType])
[error]  --- because ---
[error] argument expression's type is not compatible with formal parameter type;
[error]  found   : magnolia.Param[magnolia.examples.HasDefault,T] => scala.util.Either[String,magnolia.Param[magnolia.examples.HasDefault,T]#PType]
[error]  required: magnolia.Param[magnolia.examples.HasDefault,T] => ?Monad[?PType]
[error]     def defaultValue = ctx.constructMonadic { param =>
[error]                            ^
[error] /Users/mathias/Documents/forks/magnolia/examples/shared/src/main/scala/default.scala:31:53: type mismatch;
[error]  found   : magnolia.Param[magnolia.examples.HasDefault,T] => scala.util.Either[String,magnolia.Param[magnolia.examples.HasDefault,T]#PType]
[error]  required: magnolia.Param[magnolia.examples.HasDefault,T] => Monad[PType]
[error]     def defaultValue = ctx.constructMonadic { param =>
[error]                                                     ^
[error] /Users/mathias/Documents/forks/magnolia/examples/shared/src/main/scala/default.scala:31:45: could not find implicit value for parameter monadic: mercator.Monadic[Monad]
[error]     def defaultValue = ctx.constructMonadic { param =>
[error]                                             ^
[info] Done compiling.
[error] three errors found
```

Why hasn't travis picked this up?

**Comment by jto** [comment]
Same here. There's a type inference issue with Scala 2.11.
The fix is not trivial.

**Comment by kubukoz** [comment]
I managed to reduce 3 errors to one by enabling `-Ypartial-unification`, not sure how to proceed further...

Not sure why Travis didn't pick it up, as the test script is `sbt +compile +tests/run`, and running that locally reproduces the issue at the `+compile` step (failing to compile `examples` under 2.11.x)

**Comment by adamw** [comment]
Fixed by removing 2.11 support :)
